### PR TITLE
encrypted private grpc

### PR DESCRIPTION
**Pull request by bryanchriswhite**
Is there any specific reason not to do this?

This would add encryption to the private GRPC server. This is different from how we use TLS on the public server in that it performs no "peer certificate verification" and doesn't require a client certificate. It just uses the existing server identity's certificate on the server side.

### CODE REVIEW

File: pkg/transport/insecure.go [hunk 1]
```diff
@@ -5,19 +5,23 @@ package transport
...
+	"google.golang.org/grpc/credentials"
 )
 
-// DialAddressInsecure returns an insecure grpc connection without tls to a node.
+// DialAddressEncrypted returns a grpc connection with encryption to a node; no client auth/cert is used.
 //
 // Use this method for communication with localhost. For example, with the inspector or debugging services.
 // Otherwise in most cases DialNode should be used for communicating with nodes since it is secure.
-func DialAddressInsecure(ctx context.Context, address string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
+func DialAddressEncrypted(ctx context.Context, address string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
```
Comment (egonelbre) [inline pkg/transport/insecure.go]: This needs some better name. `DialAddressClient`, `DialAddressInspector` or something like that... `Encrypted` probably isn't clear in what the intent of the func is.

Comment (bryanchriswhite) [reply pkg/transport/insecure.go]: I get what you're saying but I'm having a hard time coming up with another name. `DialInspectorAddress` is currently the only thing it's being used for but it's not restricted to that usage. The most specific, non-usage related description is that it dials an encrypted connection.

Comment (bryanchriswhite) [reply pkg/transport/insecure.go]: What about `DialAddressEncryptedInsecure`?

Comment (jtolds) [reply pkg/transport/insecure.go]: DialAddress is already Encrypted. Maybe just DialAddressUnverified?

**Comment by jtolds** [comment]
for a localhost-only debug port, this adds no security benefit and makes connection slightly more complicated, but i can also see arguments for "if it doesn't matter why not prefer secure." in addition grpc prefers ssl, so okay, i guess it doesn't matter.

File: pkg/transport/insecure.go (hunk 1 above)
Comment (bryanchriswhite) [reply pkg/transport/insecure.go]: Good point; I like that.

File: pkg/peertls/tlsopts/tls.go [hunk 2]
```diff
@@ -22,6 +22,15 @@ func (opts *Options) ServerOption() grpc.ServerOption {
 	return grpc.Creds(credentials.NewTLS(tlsConfig))
 }
 
+// ServerNoClientAuthOption returns a grpc `ServerOption` for incoming connections
+// to the node with this full identity. As opposed to `otps.ServerOption`, does not
```
Comment (JessicaGreben) [inline pkg/peertls/tlsopts/tls.go]: spelling err: `otps.ServerOption` -> `opts.ServerOption`

**Comment by bryanchriswhite** [comment]
@jt, the thought was, why not more secure by default, and with the debug token bit that Matt was talking about this would allow a user to safely expose this externally if they wanted.

**Comment by jtolds** [comment]
oh okay, exposing externally is nice

File: pkg/peertls/tlsopts/tls.go (hunk 2 above)
Comment (bryanchriswhite) [reply pkg/peertls/tlsopts/tls.go]: nice catch :D

**Comment by bryanchriswhite** [comment]
This is waiting for some more work on the private grpc endpoint.

**Comment by mniewrzal** [comment]
@bryanchriswhite do you plan to continue developing this PR?

**Comment by bryanchriswhite** [comment]
> @bryanchriswhite do you plan to continue developing this PR?

Yes

**Pull request closed without merging**